File: openc3/python/openc3/io/stderr.py
# Copyright 2023 OpenC3, Inc.
# All Rights Reserved.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
# See LICENSE.md for more details.
#
# This file may also be used under the terms of a commercial license
# if purchased from OpenC3, Inc.

from openc3.io.io_multiplexer import IoMultiplexer


# Adds STDOUT to the multiplexed streams
class Stderr(IoMultiplexer):
    my_instance = None

    def __init__(self):
        super().__init__()
        self.streams.append(self.STDERR)
        Stderr.my_instance = self

    # @return [Stdout] Returns a single instance of Stdout
    @classmethod
    def instance(cls):
        if not Stderr.my_instance:
            cls()
        return Stderr.my_instance

    @property
    def encoding(self):
        return self.STDERR.encoding

File: openc3/lib/openc3/io/stderr.rb
# encoding: ascii-8bit

# Copyright 2022 Ball Aerospace & Technologies Corp.
# All Rights Reserved.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
# See LICENSE.md for more details.

# Modified by OpenC3, Inc.
# All changes Copyright 2026, OpenC3, Inc.
# All Rights Reserved
#
# This file may also be used under the terms of a commercial license
# if purchased from OpenC3, Inc.

require 'openc3/io/io_multiplexer'

module OpenC3
  # Adds STDERR to the multiplexed streams
  class Stderr < IoMultiplexer
    @@instance = nil

    def initialize
      super()
      @streams << STDERR
      @@instance = self
    end

    # @return [Stderr] Returns a single instance of Stderr
    def self.instance
      self.new unless @@instance
      @@instance
    end

    def tty?
      false
    end
  end
end
